homeassistant/components/shelly/button.py

```diff
@@ -0,0 +1,84 @@
+"""Button for Shelly."""
+from __future__ import annotations
+
+from typing import cast
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_DEVICE_ID, ENTITY_CATEGORY_CONFIG
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import slugify
+
+from . import BlockDeviceWrapper, RpcDeviceWrapper
+from .const import (
+    ATTR_BETA,
+    BLOCK,
+    CONF_OTA_BETA_CHANNEL,
+    DATA_CONFIG_ENTRY,
+    DOMAIN,
+    RPC,
+    SERVICE_OTA_UPDATE,
+)
+from .utils import get_block_device_name, get_device_entry_gen, get_rpc_device_name
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set buttons for device."""
+    wrapper: RpcDeviceWrapper | BlockDeviceWrapper | None = None
+    if get_device_entry_gen(config_entry) == 2:
+        if rpc_wrapper := hass.data[DOMAIN][DATA_CONFIG_ENTRY][
+            config_entry.entry_id
+        ].get(RPC):
+            wrapper = cast(RpcDeviceWrapper, rpc_wrapper)
+    else:
+        if block_wrapper := hass.data[DOMAIN][DATA_CONFIG_ENTRY][
+            config_entry.entry_id
+        ].get(BLOCK):
+            wrapper = cast(BlockDeviceWrapper, block_wrapper)
+
+    if wrapper is not None:
+        async_add_entities([ShellyOtaUpdateButton(wrapper, config_entry)])
+
+
+class ShellyOtaUpdateButton(ButtonEntity):
```

Comment:
Should the button become unavailable if there is no update pending?

Reply:
we already discussed this on discord, but decided to leave it for a future improvement PR

Reply:
I agree, let's cleanup this PR and remove all services related stuff and we can add the unavailable on a follow up PR